src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ParameterBinder.cs

```diff
@@ -279,60 +224,49 @@ public class ParameterBinder
-                // on back-compatibility logic. In this scenario, top-level validation
-                // attributes will be ignored like they were historically.
-                if (modelBindingResult.IsModelSet)
-                {
-                    _validatorForBackCompatOnly.Validate(
-                        actionContext,
-                        modelBindingContext.ValidationState,
-                        modelBindingContext.ModelName,
-                        modelBindingResult.Model);
-                }
-            }
-            else
-            {
-                Logger.AttemptingToValidateParameterOrProperty(parameter, modelBindingContext);
+            Logger.AttemptingToValidateParameterOrProperty(parameter, modelBindingContext);
 
-                EnforceBindRequiredAndValidate(
-                    actionContext,
-                    metadata,
-                    modelBindingContext,
-                    modelBindingResult);
+            Validate(actionContext, metadata, modelBindingContext, modelBindingResult);
 
-                Logger.DoneAttemptingToValidateParameterOrProperty(parameter, modelBindingContext);
-            }
+            Logger.DoneAttemptingToValidateParameterOrProperty(parameter, modelBindingContext);
 
             return modelBindingResult;
         }
 
-        private void EnforceBindRequiredAndValidate(ActionContext actionContext, ModelMetadata metadata, ModelBindingContext modelBindingContext, ModelBindingResult modelBindingResult)
+        private void Validate(
+            ActionContext actionContext,
+            ModelMetadata metadata,
+            ModelBindingContext modelBindingContext,
+            ModelBindingResult modelBindingResult)
         {
-            if (!modelBindingResult.IsModelSet && metadata.IsBindingRequired)
+            // Validate top level properties only if users inherited from this base type,
+            // otherwise fallback to earlier behavior.
+            if (_objectModelValidator is ObjectValidatorBase validator)
             {
-                // Enforce BindingBehavior.Required (e.g., [BindRequired])
-                var modelName = modelBindingContext.FieldName;
-                var message = metadata.ModelBindingMessageProvider.MissingBindRequiredValueAccessor(modelName);
-                actionContext.ModelState.TryAddModelError(modelName, message);
+                if (!modelBindingResult.IsModelSet && metadata.IsBindingRequired)
+                {
+                    // Enforce BindingBehavior.Required (e.g., [BindRequired])
+                    var modelName = modelBindingContext.FieldName;
+                    var message = metadata.ModelBindingMessageProvider.MissingBindRequiredValueAccessor(modelName);
+                    actionContext.ModelState.TryAddModelError(modelName, message);
+                }
+                else if (modelBindingResult.IsModelSet || metadata.IsRequired)
```

Comment:
Confusing. Won't everything be just fine with just `else`? Let the `ValidationVisitor` no-op.